core/src/main/java/org/apache/gravitino/authorization/AuthorizationUtils.java

```diff
@@ -131,58 +132,127 @@ public static <T> void prepareAuthorizationHooks(T manager, DispatcherHooks hook
       hooks.addPostHook(
           "createMetalake",
           (args, metalake) -> {
-            // TODO: Add the logic of setting the owner
+            NameIdentifier identifier = (NameIdentifier) ((Object[]) args)[0];
```

Comment:
It's a little hacky to get instance of NameIdentifier here, how can I know the type of the first argument is `NameIdentifier`?

Reply:
You can know it through the method `createMetalake`.

Reply:
So if we change the method `createMetalake` and the type of first argument is NOT `NameIdentifier`,  we could not detect the error through compilation stage.

Reply:
But we can add some e2e test to detect it.